content/frontend-web/react-typescript/design-pattern.md

```diff
@@ -6,6 +6,217 @@ showToc: true
 order: 2
 ---
 
+# Import & Export (JavaScript modules)
+
+- Default export
+
+    ```ts
+    // MyComponent.tsx
+    const MyComponent = () => <div>Hey, MyComponent</div>;
+    export default MyComponent;
+    // App.tsx
+    import MyComponent from './MyComponent';
+    ```
+
+- Export
+
+    ```ts
+    // MyComponent.tsx
+    export const MyComponent = () => <div>Hey, MyComponent</div>;
+    // App.tsx
+    import { MyComponent } from './MyComponent';
+    ```
+
+- Rename import
+
+    ```ts
+    // MyComponent.tsx
+    export const MyComponent = () => <div>Hey, MyComponent</div>;
+    // App.tsx
+    import { MyComponent as MySuperComponent } from './MyComponent';
+    export const App = () => <MySuperComponent />;
+    ```
+
+- Group all imports and rename
+
+    ```tsx
+    // MyComponent.tsx
+    export const Switch = () => <div>Hey, Switch</div>;
+    export const Group = () => <div>Hey, Group</div>;
+    // App.tsx
+    import { * as MyComponent } from './MyComponent';
+    export const App = () => (
+        <>
+            <MyComponent.Switch />
+            <MyComponent.Group />
+        </>
+    );
+    ```
+
+- Re-export all
+
+    ```ts
+    // MyComponent.tsx
+    export const Switch = () => <div>Hey, Switch</div>;
+    export const Group = () => <div>Hey, Group</div>;
+    // MyParentComponent.tsx
+    export * from "./MyComponent";
+    // App.tsx
+    import { Switch, Group } from './MyComponent';
+    export const App = () => (
+        <>
+            <Switch />
+            <Group />
+        </>
+    );
+    ```
+
+- Re-export, then group all imports and rename
+
+    ```ts
+    // MyComponent.tsx
+    export const Switch = () => <div>Hey, Switch</div>;
+    export const Group = () => <div>Hey, Group</div>;
+    // MyParentComponent.tsx
+    export  * as MyComponent from "./MyComponent";
+    // App.tsx
+    import { MyComponent } from './MyComponent';
+    export const App = () => (
+        <>
+            <MyComponent.Switch />
+            <MyComponent.Group />
+        </>
+    );
+    ```
+
+
+- Export type
+
+    ```ts
+    // MyComponent.tsx
+    export type Sex = "Male" | "Female"; // Accept only "Male" or "Female" 

```

Comment:
Just a minor change, it think when taking about male or female we prefer to use `gender`.
https://www.oxfordlearnersdictionaries.com/definition/english/gender?q=gender

Reply:
Good word.